### DISCUSSION TITLE
CLI: regex in string_fragment

### DISCUSSION THREAD

**Discussion by Isos9** [ast-grep/ast-grep#1213] [Q&A]
Hi,

This library is great.

I am testing and was wondering if it's possible with CLI only (no YAML) to narrow the search I do with regex in string_fragment nodes. for example `console.log('\w value::', $VAL)`, or `console.log('\w =>', $$$)`

In this case regex would be a good use case for a quick search in my opinion, instead of creating a rule in YAML.

Unless I can enhance my quick searches with YAML to add this kind of rule, in this case, I would like to understand how it's possible.
To me it seems to be used for lining only at the moment.

Thanks!

**Comment by Isos9** [comment]
Just saw this: https://github.com/ast-grep/ast-grep/discussions/378

So I suppose it's not supported.

I believe that would make a real difference if regex can be injected in a pattern with the syntax mentioned in this discussion or similar, that would give a better experience for search plugins for vscode and nvim

**Reply by HerringtonDarkholme** [reply]
You can use https://ast-grep.github.io/reference/cli/scan.html#inline-rules-rule-text `--inline-rules` to use YAML on CLI. 
alternatively, `sg run` can also have metavar filter in the future. https://github.com/ast-grep/ast-grep/issues/379
